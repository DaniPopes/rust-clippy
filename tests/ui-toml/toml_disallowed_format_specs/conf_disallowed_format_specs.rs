#![warn(clippy::disallowed_format_specs)]

pub struct Address;

macro_rules! impl_fmt {
    ($($t:ident),*) => {$(
        impl std::fmt::$t for Address {
            fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
                unimplemented!()
            }
        }
    )*}
}

impl_fmt!(Display, Debug, LowerHex, UpperHex);

fn main() {
    let x = &&(&&(Address));

    let _ = format!("{}", Address);
    let _ = format!("{:#}", Address);
    let _ = format!("{:?}", Address);
    //~^ ERROR: format trait `Debug` is not allowed for type `Address` according to config
    let _ = format!("{:x}", Address);
    //~^ ERROR: format trait `LowerHex` is not allowed for type `Address` according to config
    let _ = format!("{:#x}", Address);
    //~^ ERROR: format trait `LowerHex` is not allowed for type `Address` according to config
    let _ = format!("{:X}", Address);
    //~^ ERROR: format trait `UpperHex` is not allowed for type `Address` according to config
    let _ = format!("{:#X}", Address);
    //~^ ERROR: format trait `UpperHex` is not allowed for type `Address` according to config

    let _ = format!("{}", x);
    let _ = format!("{:#}", x);
    let _ = format!("{:?}", x);
    //~^ ERROR: format trait `Debug` is not allowed for type `Address` according to config
    let _ = format!("{:x}", x);
    //~^ ERROR: format trait `LowerHex` is not allowed for type `Address` according to config
    let _ = format!("{:#x}", x);
    //~^ ERROR: format trait `LowerHex` is not allowed for type `Address` according to config
    let _ = format!("{:X}", x);
    //~^ ERROR: format trait `UpperHex` is not allowed for type `Address` according to config
    let _ = format!("{:#X}", x);
    //~^ ERROR: format trait `UpperHex` is not allowed for type `Address` according to config

    let _ = format!("{x}");
    let _ = format!("{x:#}");
    let _ = format!("{x:?}");
    //~^ ERROR: format trait `Debug` is not allowed for type `Address` according to config
    let _ = format!("{x:x}");
    //~^ ERROR: format trait `LowerHex` is not allowed for type `Address` according to config
    let _ = format!("{x:#x}");
    //~^ ERROR: format trait `LowerHex` is not allowed for type `Address` according to config
    let _ = format!("{x:X}");
    //~^ ERROR: format trait `UpperHex` is not allowed for type `Address` according to config
    let _ = format!("{x:#X}");
    //~^ ERROR: format trait `UpperHex` is not allowed for type `Address` according to config
}
