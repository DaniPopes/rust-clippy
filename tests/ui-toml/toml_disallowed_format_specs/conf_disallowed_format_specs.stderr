error: format trait `Debug` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:22:29
   |
LL |     let _ = format!("{:?}", Address);
   |                             ^^^^^^^
   |
   = note: `-D clippy::disallowed-format-specs` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::disallowed_format_specs)]`

error: format trait `LowerHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:24:29
   |
LL |     let _ = format!("{:x}", Address);
   |                             ^^^^^^^

error: format trait `LowerHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:26:30
   |
LL |     let _ = format!("{:#x}", Address);
   |                              ^^^^^^^

error: format trait `UpperHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:28:29
   |
LL |     let _ = format!("{:X}", Address);
   |                             ^^^^^^^

error: format trait `UpperHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:30:30
   |
LL |     let _ = format!("{:#X}", Address);
   |                              ^^^^^^^

error: format trait `Debug` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:35:29
   |
LL |     let _ = format!("{:?}", x);
   |                             ^

error: format trait `LowerHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:37:29
   |
LL |     let _ = format!("{:x}", x);
   |                             ^

error: format trait `LowerHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:39:30
   |
LL |     let _ = format!("{:#x}", x);
   |                              ^

error: format trait `UpperHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:41:29
   |
LL |     let _ = format!("{:X}", x);
   |                             ^

error: format trait `UpperHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:43:30
   |
LL |     let _ = format!("{:#X}", x);
   |                              ^

error: format trait `Debug` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:48:23
   |
LL |     let _ = format!("{x:?}");
   |                       ^

error: format trait `LowerHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:50:23
   |
LL |     let _ = format!("{x:x}");
   |                       ^

error: format trait `LowerHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:52:23
   |
LL |     let _ = format!("{x:#x}");
   |                       ^

error: format trait `UpperHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:54:23
   |
LL |     let _ = format!("{x:X}");
   |                       ^

error: format trait `UpperHex` is not allowed for type `Address` according to config
  --> $DIR/conf_disallowed_format_specs.rs:56:23
   |
LL |     let _ = format!("{x:#X}");
   |                       ^

error: aborting due to 15 previous errors

